File: change-maker/typescript/src/changeMaker.ts
export function makeChange(amount: number, denominations: readonly number[]): number[] {
  const coins: number[] = [];
  let remaining = amount;
  for (const coin of denominations) {
    while (remaining >= coin) {
      coins.push(coin);
      remaining -= coin;
    }
  }
  return coins;
}

File: change-maker/csharp/src/ChangeMaker/ChangeMaker.cs
namespace ChangeMaker;

public static class ChangeMaker
{
    public static IReadOnlyList<int> MakeChange(int amount, IReadOnlyList<int> denominations)
    {
        var coins = new List<int>();
        var remaining = amount;
        foreach (var coin in denominations)
        {
            while (remaining >= coin)
            {
                coins.Add(coin);
                remaining -= coin;
            }
        }
        return coins;
    }
}
